core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/UserUniqueIDDomainResolver.java

```diff
@@ -209,6 +212,7 @@ private void persistDomainAgainstUserId(String userId, String userDomain, int te
                     preparedStatement.setInt(3, tenantId);
                     preparedStatement.setInt(4, tenantId);
                     preparedStatement.execute();
+                    commitTransaction(dbConnection);
                 }
             }
         } catch (SQLException ex) {
```

Comment:
We need to handle the rollback as well.

Reply:
Rollback handled https://github.com/wso2/carbon-kernel/pull/2770/files#diff-64f4ebd98c48b6db56341e6f89eb219dR253 in here

Reply:
But there can be SQLExceptions before commit to the commitTransaction..

Reply:
done